test/python/visualization/test_circuit_matplotlib_drawer.py

```diff
@@ -52,37 +48,27 @@ def _expected_empty(self):
         expected.set_size_inches(2.508333333333333, 0.2508333333333333)
         return expected
 
-    def _make_temp_file(self, plot):
-        tmp = tempfile.NamedTemporaryFile(suffix='.png')
-        self.addCleanup(tmp.close)
-        plot.savefig(tmp.name)
-        return tmp
-
     @unittest.skipIf(not visualization.HAS_MATPLOTLIB,
                      'matplotlib not available.')
-    @unittest.skipIf(os.name == 'nt', 'tempfile fails on appveyor')
     def test_empty_circuit(self):
         qc = QuantumCircuit()
-        res = visualization.circuit_drawer(qc, output='mpl')
-        res_out_file = self._make_temp_file(res)
+        filename = self._get_resource_path('current_pulse_matplotlib_ref.png')
+        visualization.circuit_drawer(qc, output='mpl', filename=filename)
+
+        expected_filename = self._get_resource_path('expected_current_pulse_matplotlib_ref.png')
         expected = self._expected_empty()
-        expected_image_file = self._make_temp_file(expected)
-        self.assertIsNone(compare.compare_images(expected_image_file.name,
-                                                 res_out_file.name, 0.0001))
+        expected.savefig(expected_filename)
+
+        self.assertImagesAreEqual(filename, expected_filename)
+        os.remove(filename)
+        os.remove(expected_filename)
```

Comment:
We should use `self.addCleanup()` for all the `os.remove()` calls. Without it if the test fails for some reason we don't ever run this it will leave dangling files.